File: aoc/src/main/java/day1/DAY1.java
package main.java.day1;

import java.util.ArrayList;
import java.util.List;

import static main.java.common.Util.readfile;

public class DAY1 {
    static int result = 0;
    static List<Integer> frequencies = new ArrayList<>();
    private static int reocurringFrequency = 0;

    public static void main(String[] args) {

        // same input for bot parts of the challenge
        final String path = "/Users/laszboga/learning/adventoOfCode/aoc/src/main/java/day1/input1_1.txt";
        final List<String> lines = readfile(path);
        lines.forEach(DAY1::calculateSumFromLine);
        while(reocurringFrequency == 0 )  {
            lines.forEach(l -> addAndCheckFrequency(calculateSumFromLine(l)));
        }
        System.out.println("reocurringFrequency = " + reocurringFrequency);
    }

    private static void addAndCheckFrequency(int frequency) {
        if (!frequencies.contains(frequency)) {
            frequencies.add(frequency);
        } else {
            System.out.println(frequency);
            reocurringFrequency = frequency;
        }
    }

    private static int calculateSumFromLine(String l) {
        result += Integer.parseInt(l);
        return result;
    }
}

File: aoc/src/main/java/twentytwenty/day1/DAY1.kt
package main.java.twentytwenty.day1

import main.java.common.Util

fun main() {
    // same input for bot parts of the challenge


    // same input for bot parts of the challenge
//    val path = "/Users/laszboga/learning/adventoOfCode/aoc/src/main/java/day1/input1_1.txt"
    val path = "/Users/laszboga/learning/adventoOfCode/aoc/src/main/java/twentytwenty/day1/input_day1_1.txt"
    val lines = Util.readfile(path)
    val day1 = DAY1(lines.map { it.toInt() })
    println(day1.getResult(2020))
    println(day1.getResult2(2020))
}

class DAY1(private val inputs: List<Int>) {
    fun getResult(target: Int): Int {
        val valueSet: MutableSet<Int> = mutableSetOf()

        for (value in inputs) {
            if (valueSet.contains(target - value)) {
                return value * (target - value);
            }
            valueSet.add(value)
        }
        return 0;
    }

    fun getResult2(target: Int): Int {
        for ((i, a) in inputs.withIndex()) {
            for (j in i + 1 until inputs.size - 2) {
                for (k in j + 1 until inputs.size - 1) {
                    if (a + inputs[j] + inputs[k] == target) return a * inputs[j] * inputs[k]

                }
            }
        }
        return 0;

    }
}